lib/internal/eachOfLimit.js

```diff
@@ -50,5 +50,6 @@ export default function _eachOfLimit(limit) {
         }
 
         replenish();
+        return callback.promise;
```

Comment:
I think we'll want to track whether the callback is a user defined callback or an internal promise callback and only return `promiseCallback.promise` if its an internal version to avoid cases where users pass us a callback with some state on it.